src/plugins/ftp/include/plugins/ftp/ftp.h

```diff
@@ -139,6 +139,17 @@ class Ftp : public PluginBase {
      */
     void upload_async(std::string local_file_path, std::string remote_dir, UploadCallback callback);
 
+    /**
+     * @brief Callback type for new file uploaded by remote.
+     */
+
+    using FileUploadedCallback = std::function<void(std::string)>;
+
+    /**
+     * @brief Register a callback to be triggered when remote uploads a file to FTP server
+     */
+    void register_file_uploaded_callback(FileUploadedCallback callback);
+
```

Comment:
This file is auto-generated from https://github.com/mavlink/MAVSDK-Proto/blob/main/protos/ftp/ftp.proto, you can't change it like that.